api/src/main/java/org/openmrs/module/referenceapplication/ReferenceApplicationActivator.java

```diff
@@ -113,18 +120,36 @@ public void setupConceptManagementAppsGlobalProperties(AdministrationService adm
 		    "1ADDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD");
 	}
 
-    public void setupEmrApiGlobalProperties(AdministrationService administrationService) {
-        setGlobalProperty(administrationService, EmrApiConstants.GP_ADMISSION_ENCOUNTER_TYPE, ReferenceMetadataProperties.ADMISSION_ENCOUNTER_TYPE_UUID);
-        setGlobalProperty(administrationService, EmrApiConstants.GP_TRANSFER_WITHIN_HOSPITAL_ENCOUNTER_TYPE, ReferenceMetadataProperties.TRANSFER_ENCOUNTER_TYPE_UUID);
-        setGlobalProperty(administrationService, EmrApiConstants.GP_EXIT_FROM_INPATIENT_ENCOUNTER_TYPE, ReferenceMetadataProperties.DISCHARGE_ENCOUNTER_TYPE_UUID);
-        setGlobalProperty(administrationService, EmrApiConstants.GP_CHECK_IN_ENCOUNTER_TYPE, ReferenceMetadataProperties.CHECK_IN_ENCOUNTER_TYPE_UUID);
+    public void setupEmrApiGlobalProperties(AdministrationService administrationService, MetadataMappingService metadataMappingService) {
+		MetadataSource emrapiSource = metadataMappingService.getMetadataSourceByName(EmrApiConstants.EMR_METADATA_SOURCE_NAME);
 
-        setGlobalProperty(administrationService, EmrApiConstants.GP_AT_FACILITY_VISIT_TYPE, ReferenceMetadataProperties.FACILITY_VISIT_TYPE_UUID);
+		EncounterType admissionEncounterType = Context.getEncounterService().getEncounterTypeByUuid(ReferenceMetadataProperties.ADMISSION_ENCOUNTER_TYPE_UUID);
+		mapMetadata(metadataMappingService, emrapiSource, EmrApiConstants.GP_ADMISSION_ENCOUNTER_TYPE, admissionEncounterType);
+
+		EncounterType transferEncounterType = Context.getEncounterService().getEncounterTypeByUuid(ReferenceMetadataProperties.TRANSFER_ENCOUNTER_TYPE_UUID);
+		mapMetadata(metadataMappingService, emrapiSource, EmrApiConstants.GP_TRANSFER_WITHIN_HOSPITAL_ENCOUNTER_TYPE, transferEncounterType);
+
+		EncounterType exitEncounterType = Context.getEncounterService().getEncounterTypeByUuid(ReferenceMetadataProperties.DISCHARGE_ENCOUNTER_TYPE_UUID);
+		mapMetadata(metadataMappingService, emrapiSource, EmrApiConstants.GP_EXIT_FROM_INPATIENT_ENCOUNTER_TYPE, exitEncounterType);
+
+		EncounterType checkInEncounterType = Context.getEncounterService().getEncounterTypeByUuid(ReferenceMetadataProperties.CHECK_IN_ENCOUNTER_TYPE_UUID);
+		mapMetadata(metadataMappingService, emrapiSource, EmrApiConstants.GP_CHECK_IN_ENCOUNTER_TYPE, checkInEncounterType);
+
+		VisitType facilityVisitType = Context.getVisitService().getVisitTypeByUuid(ReferenceMetadataProperties.FACILITY_VISIT_TYPE_UUID);
+		mapMetadata(metadataMappingService, emrapiSource, EmrApiConstants.GP_AT_FACILITY_VISIT_TYPE, facilityVisitType);
 
         setGlobalProperty(administrationService, EmrApiConstants.GP_DIAGNOSIS_SET_OF_SETS, ReferenceMetadataProperties.ICPC_DIAGNOSIS_CATEGORIES_CONCEPT_UUID);
```

Comment:
Did you leave out the above line intentionally? Or could you explain why you left it as a global property?